### PR TITLE
Pr network connections

### PR DESCRIPTION
**Pull request by chrehall68**
* only includes relevant qt-6 upgrade changes
* add helper methods so that code is much more organized
* fix bug where all networks were seen as "no security"
* fix bug where the OtherNetworkWidget didn't communicate password with network manager (entering the password on OtherNetworkWidget worked, but the system connection didn't get that). Resolves kipr/botui#117 
* organize connecting to networks
* add comments
* Fix bug where botui doesn't display the password of the current network connection in the Network Information area

### CODE REVIEW

File: src/NetworkManager.cpp [hunk 1]
```diff
@@ -320,135 +239,55 @@ void NetworkManager::turnOff()
...
   DEFAULT_AP["ipv4"]["method"] = "shared";
   DEFAULT_AP["ipv6"]["method"] = "auto";
   DEFAULT_AP["connection"]["type"] = "802-11-wireless";
   DEFAULT_AP["connection"]["uuid"] = QUuid::createUuid().toString().remove('{').remove('}');
   // File name is just the SSID for now
-  DEFAULT_AP["connection"]["id"] = "APName";
-  DEFAULT_AP["connection"]["autoconnect"] = false;
-  DEFAULT_AP["connection"]["autoconnect-priority"] = -900;
-  DEFAULT_AP["connection"]["interface-name"] = "wlo1";
+  DEFAULT_AP["connection"]["id"] = AP_NAME;
```
Comment (tcorbly) [inline src/NetworkManager.cpp]: Add to new dynamic AP wombat name PR.

File: src/NetworkManager.cpp [hunk 2]
```diff
@@ -67,6 +68,18 @@
...
 Connection DEFAULT_AP;
 
+#ifndef WOMBAT
+#define WIFI_DEVICE "wlp2s0" // varies per pc, almost always need to change this on different computers
+#define AP_NAME "COOL_NETWORK"
+#define AP_SSID QByteArray("COOL_NETWORK")
+#define AP_PASSWORD "COOL_PASSWORD"
+#else
+#define WIFI_DEVICE "wlan0" // always wlan0 for raspberry pi
+#define AP_NAME m_dev->serial() + "-wombat"
```
Comment (tcorbly) [inline src/NetworkManager.cpp]: Add to new dynamic AP wombat name PR.

File: src/NetworkManager.cpp [hunk 3]
```diff
@@ -67,6 +68,18 @@
...
+#define AP_SSID (m_dev->serial() + "-wombat").toUtf8()
```
Comment (tcorbly) [inline src/NetworkManager.cpp]: Add to new dynamic AP wombat name PR.

File: src/main.cpp [hunk 4]
```diff
@@ -20,45 +20,45 @@
...
+  QFontDatabase::addApplicationFont(":/fonts/DejaVuSansMono.ttf");
+
+  srand(time(NULL));
+
+  Wombat::Device device;
+  CursorManager::ref().setDevice(&device);
 #ifdef QT_DBUS_LIB
   KovanSerialBridge::ref().init(&device);
-  NetworkManager::ref();
+  NetworkManager::ref().init(&device);
```
Comment (tcorbly) [inline src/main.cpp:53]: Add to new dynamic AP wombat name PR.

**Comment by chrehall68** [comment]
Since this has become stale, I'm breaking this down into separate pr's and will close this once those have been made (or this has been merged, whichever comes first)

**Pull request closed without merging**